Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScopeTests.swift

```diff
@@ -36,6 +37,26 @@ class RUMResourceScopeTests: XCTestCase {
         XCTAssertEqual(scope.context.activeUserActionID, try XCTUnwrap(context.activeUserActionID))
     }
 
+    func testWhenScopeIsCreated_itReceivesRandomUUID() {
+        func createScope() -> RUMResourceScope {
+            RUMResourceScope(
+                context: context,
+                dependencies: .mockAny(),
+                resourceName: .mockAny(),
+                attributes: [:],
+                startTime: .mockAny(),
+                url: .mockAny(),
+                httpMethod: .mockAny(),
+                spanContext: nil
+            )
+        }
+
+        let scope1 = createScope()
+        let scope2 = createScope()
+
+        XCTAssertNotEqual(scope1.resourceUUID, scope2.resourceUUID)
```

Comment:
i think we shouldn't be testing this, we need `id`s in generated `RUMResourceResource` to be different, no?
`resourceUUID` can be hidden as `private` since it is an implementation detail; the actual identifier of this type is `resourceName`

Reply:
Hmm, not sure if I understand. Each Resource should have unique `UUID`, why we shouldn't test it?

Reply:
yes, i mean each `Resource` (the output of `ResourceScope`) must have unique identifiers. here we are testing identifiers of `ResourceScope` only. 
does that make sense?

Reply:
It was testing uniqueness of the `scope.id`, and later that `event.id == scope.id`. Anyway, we can easily make assertions on the actual events sent, so I updated it 👍. Also, added more validation logic in `RUMSessionMatcher`, so we have this assertion for free in all integration tests.